tests/test_rtdc_feat_anc_plugin.py

```diff
@@ -903,7 +904,88 @@ def test_pf_wrong_length_2():
         with pytest.warns(BadFeatureSizeWarning,
                           match="to match event number"):
             ds[pf.feature_name]
+            
 
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_non_scalar_plugin_data_with_shape_issue_171():
+    """Test storage of 3D image plugin feature attributes with shape info
+    in recipe"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_non_scalar_feature()
+    # Adding feature shape to the info
+    info["feature shapes"] = [(80, 250)]
+    PlugInFeature("image_gauss_filter", info)
+    with dclab.new_dataset(h5path) as ds:
+        image_gauss_filter = gaussian_filter(ds["image"], sigma=(0, 1, 1))
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("image_gauss_filter", image_gauss_filter)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "image_gauss_filter" in events.keys()
+        feat = h5["events"]['image_gauss_filter']
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_non_scalar_plugin_data_without_shape_issue_171():
+    """Test storage of 3D image plugin feature attributes without shape info
+     in recipe"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_non_scalar_feature()
+    PlugInFeature("image_gauss_filter", info)
+    with dclab.new_dataset(h5path) as ds:
+        image_gauss_filter = gaussian_filter(ds["image"], sigma=(0, 1, 1))
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("image_gauss_filter", image_gauss_filter)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "image_gauss_filter" in events.keys()
+        feat = h5["events"]['image_gauss_filter']
+        assert b'IMAGE' in feat.attrs.get('CLASS')
```

Comment:
for all these tests, we want to assert that the shape is as expected in the reloaded hdf5 file, so please check for this.

Reply:
Yes, I have added but haven't committed the changes.

Reply:
Did you do this??

Reply:
Yes, you can find them like
```
fshape = feat.shape
assert feat.ndim == 3
```

Reply:
I don't see it here, please link the line of code

Reply:
Didn't you mean line number 1002? (check the shape of the feature)

Reply:
I have no idea what line of code it is FYI, I just want you to post a link to the solution.